crates/script/src/lib.rs

```diff
@@ -294,9 +303,10 @@ impl ScriptArgs {
             pre_simulation.args.check_contract_sizes(
                 &pre_simulation.execution_result,
                 &pre_simulation.build_data.known_contracts,
+                create2_deployer,
             )?;
 
-            pre_simulation.fill_metadata().await?.bundle().await?
+            pre_simulation.fill_metadata(create2_deployer).await?.bundle().await?
```

Comment:
`pre_simulation` can use `script_config.evm_opts.create2_deployer`, so we don't need to pass it explicitly here and in some other places

Reply:
thanks for this